package bbm;

message BBMUserEvent {

	required int64 timestamp = 1;
	required string uuid = 2;
	optional string apikey = 3;
	optional int32 userId = 4; //best to be optional
	required string eventType = 5;

	enum NetworkStatus {
	     ONLINE = 0;
	     OFFLINE = 1;
	}

	required NetworkStatus networkStatus = 6;

	message KeyValue {
		required string key = 1;
		required string value = 2;
	}

	message Context {
		optional string containerName = 1;
		optional string stationIdentPlaying = 2;
		optional int32 trackIdPlaying = 3;
		optional int32 trackVersionPlaying = 4;
		repeated KeyValue otherContext = 5;
	}

	required Context context = 7;

	message EventData {
		optional string stationIdent = 1;
		optional string trackId = 2;
		optional string trackVersionId = 3;
		repeated KeyValue otherEventData = 4;
	}

	required EventData eventData = 8;
}

message BBMUserEventBatch {
    repeated BBMUserEvent userEvent = 1;
}